packages/apollo/src/commands/service/push.ts

```diff
@@ -24,8 +24,9 @@ export default class ServicePush extends ProjectCommand {
     federated: flags.boolean({
       char: "f",
       default: false,
+      hidden: true,
       description:
-        "Indicates that the schema is a partial schema from a federated service"
+        "[Deprecated: use --serviceName to indicate federation] Indicates that the schema is a partial schema from a federated service"
```

Comment:
As above, can we log a deprecation notice for the federated flag at the beginning of this command?

```js
if (flags.federated) {
  this.log("The --federated flag is no longer required when running federated commands. Use of the flag will not be supported in future versions of the CLI.");
}
```